src/EcoSurveyComponent.java

```diff
@@ -0,0 +1,147 @@
+import components.map.Map;
+import components.map.Map1L;
+import components.simplereader.SimpleReader;
+import components.simplereader.SimpleReader1L;
+import components.simplewriter.SimpleWriter;
+import components.simplewriter.SimpleWriter1L;
+
+public class EcoSurveyComponent {
+    private final class Species {
+
+        //Should these fields be public?
+        /**
+         * colloquial name
+         */
+        private String name;
+        /**
+         * scientific name
+         */
+        private String trueName;
+        /**
+         * ideal population density
+         */
+        private double density;
+
+    }
+
+    //is this redundant. Should I just create another private field for count?
+    private Map<Species, Integer> speciesCount;
```

Comment:
Keep in mind that keys should be hashable. If not, there's no telling that two of the same species will be stored in the same place in the Map. I would personally store count in the Species object, and then store the species in the map by name. That way, they can be looked up quickly and easily. 
```suggestion
    private Map<String, Species> species;
```